src/hacking/building-for-openharmony.md

```diff
@@ -4,189 +4,29 @@
-care to install the hvigor version matching the requirements of your project.
-</div>
-
-`hvigor` (not the wrapper `hvigorw`) is also available via `npm`.
-1. Install the same nodejs version as the commandline-tools ship.
-   For HarmonyOS NEXT Node 18 is shipped. Ensure that the `node` binary is in PATH.
-2. Install Java using the recommended installation method for your OS.
-   The build steps are known to work with OpenJDK v17, v21 and v23.
-   On MacOS, if you install Homebrew's [OpenJDK formula], the following additional command may need to be run after the installation:
-   ```
-   # For the system Java wrappers to find this JDK, symlink it with
-   sudo ln -sfn $HOMEBREW_PREFIX/opt/openjdk/libexec/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk.jdk
-   ```
-2. Edit your `.npmrc` to contain the following line:
-
-    ```
-    @ohos:registry=https://repo.harmonyos.com/npm/
-    ```
-
-3. Install hvigor and the hvigor-ohos-plugin. This will create a `node_modules` directory in the current directory.
-   ```commandline
-   npm install @ohos/hvigor
-   npm install @ohos/hvigor-ohos-plugin
-   ```
-4. Now you should be able to run `hvigor.js` in your OpenHarmony project to build a hap bundle:
-   ```
-   /path/to/node_modules/@ohos/hvigor/bin/hvigor.js assembleHap
-   ```
-[OpenJDK formula]: https://formulae.brew.sh/formula/openjdk#default
-
-### Configuring hdc on Linux
 
-`hdc` is the equivalent to `adb` for OpenHarmony devices.
-You can find it in the `toolchains` directory of your SDK.
-For convenience purposes, you might want to add `toolchains` to your `PATH`.
-Among others, `hdc` can be used to open a shell or transfer files between a device and the host system.
-`hdc` needs to connect to a physical device via `usb`, which requires the user has permissions to access the device.
-
-It's recommended to add a `udev` rule to allow hdc to access the corresponding device without needing to run `hdc` as root.
-[This stackoverflow answer](https://stackoverflow.com/a/53887437) also applies to `hdc`.
-Run `lsusb` and check the vendor id of your device, and then create the corresponding `udev` rule.
-Please note that your user should be a member of the group you specify with `GROUP="xxx"`.
-Depending on your Linux distributions you may want to use a different group.
-
-To check if `hdc` is now working, you can run `hdc list targets` and it should show your device serial number.
-If it doesn't work, try rebooting.
-
-Please note that your device needs to be in "Developer mode" with USB debugging enabled.
-The process here is exactly the same as one android:
-1. Tap the build number multiple times to enable developer mode.
-2. Then navigate to the developer options and enable USB debugging.
-3. When you connect your device for the first time, confirm the pop-up asking you if you want to trust the computer you are connecting to.
-
-## Signing configuration
```

Comment:
Also necessary for Linux, and recommend for windows / mac too, since using ./mach install is just very convenient.